### PR TITLE
Work on dvd, si, os and exi

### PR DESCRIPTION
**Pull request by hatal175**

### CODE REVIEW

File: libs/dolphin/dvd/dvdfs.c [hunk 1]
```diff
@@ -80,173 +84,413 @@ SECTION_DATA static char lit_119[] =
...
+    const char* extentionStart;
+    BOOL illegal;
+    BOOL extention;
+
+    dirLookAt = currentDirectory;
+
+    while (1) {
+
+        if (*pathPtr == '\0') {
+        return (s32)dirLookAt;
```
Comment (TakaRikka) [inline libs/dolphin/dvd/dvdfs.c:136]: can clang format be run on some of these? this function in particular is hard to read currently

File: libs/dolphin/dvd/dvdfs.c [hunk 2]
```diff
@@ -80,173 +84,413 @@ SECTION_DATA static char lit_119[] =
...
-#include "asm/dolphin/dvd/dvdfs/DVDReadPrio.s"
+int DVDReadPrio(DVDFileInfo* fileInfo, void* addr, s32 length, s32 offset, s32 prio) {
+    BOOL result;
+    DVDCommandBlock* block;
+    s32 state;
+    BOOL enabled;
+    int retVal;
+
+    if (!((0 <= offset) && (offset <= fileInfo->length))) {
+        OSPanic(lit_118, 820, lit_265);
```
Comment (TakaRikka) [inline libs/dolphin/dvd/dvdfs.c:405]: are these literals able to be removed with the function being decompiled?

**Review by TakaRikka — COMMENTED** [review]
just a few minor things

File: libs/dolphin/dvd/dvdfs.c (hunk 2 above)
Comment (hatal175) [reply libs/dolphin/dvd/dvdfs.c:402]: Probably?

File: libs/dolphin/exi/Makefile [hunk 3]
```diff
@@ -10,16 +10,18 @@ LIBEXI_A_O_FILES := \
...
 	-fp hard \
+	-fp_contract on \
 	-nodefaults \
 	-str reuse \
+	-inline auto \
 	-RTTI off \
 	-maxerrors 5 \
 	-enum int $(INCLUDES) \
-	-O4,p \
+	$(if $(filter %EXIBios.c, $<),-O3,-O4),p \
```
Comment (jdflyer) [inline libs/dolphin/exi/Makefile:24]: We should probably use the optimization_level pragma inside of EXIBios.c instead of doing this

Comment (hatal175) [reply libs/dolphin/exi/Makefile:24]: I tried but doing #pragma optimization_level 3 didn't work for some reason

Comment (TakaRikka) [reply libs/dolphin/exi/Makefile:24]: you have to use the pragma push / pop as well in my experience at least

**Comment by hatal175** [comment]
I did. Not sure what I did wrong.

File: libs/dolphin/exi/Makefile (hunk 3 above)
Comment (hatal175) [reply libs/dolphin/exi/Makefile:24]: I did. Not sure what I did wrong.

**Review by TakaRikka — APPROVED** [review]

**Pull request merged**